### ISSUE TITLE
Detect vpn service is not launched in runtime and fallback to DoH

### ISSUE THREAD

**Issue by spylogsster** [brave/brave-browser#28195]
## Test plan
See https://github.com/brave/brave-core/pull/17013/

## Description
If brave vpn helper service was not launched by some reasons we have to detect it in runtime and fallback to DoH to protect users from DNS leaks

**Comment by LaurenWags** [comment]
@spylogsster @bsclifton is this Windows only?

**Comment by spylogsster** [comment]
> @spylogsster @bsclifton is this Windows only?

yes, added label

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using

Brave | 1.50.99 Chromium: 111.0.5563.64 (Official Build) beta (64-bit)
-- | --
Revision | c710e93d5b63b7095afe8c2c17df34408078439d-refs/branch-heads/5563@{#995}
OS | Windows 11 Version 21H2 (Build 22000.1696)

### Steps:
1. installed `1.50.99`
2. launched Brave
3. restarted (to pick up Griffin changes and enable `VPN` button)
4. loaded `account.brave.com`
5. logged in using my `stephen-prod@mailinator.com` account
6. clicked on `Refresh Brave VPN` button to obtain new credentials
7. clicked on the `VPN` button in the browser toolbar
8. toggled from `Disconnected` to `Connected`
9. confirmed I saw the modal dialog about Windows' `Smart Multi-Homed Name Resolution` and our Brave-helper-service workaround
9. pressed `ctrl` + `alt` + `del` to open `Task Manager`
10. context-clicked on the `Name` column and checked `Command line`
11. looked at the list of `Apps` and `Background processes`
12. confirmed I saw no `Brave VPN Helper` service (`brave_vpn_helper.exe` in the background-processes list
13. loaded `https://www.browserleaks.com/dns`
14. compared results to those using the helper service

![image](https://user-images.githubusercontent.com/387249/227317410-3b748238-c473-4dcb-8e1f-8d658dae6231.png)

**Comment by stephendonner** [comment]
Leaving `QA/Pass-Win64` as this should also be tested on `Windows 10`.

**Comment by GeetaSarvadnya** [comment]
Verification INPROGRESS on

```
Brave | 1.50.101 Chromium: 111.0.5563.116 (Official Build) beta (64-bit)
-- | --
Revision | ade86b9472831e26d87fd569e04b404be01e1e2b-refs/branch-heads/5563@{#1209}
OS | Windows 10 Version 21H2 (Build 19044.2728)

```

**Comment by GeetaSarvadnya** [comment]
Verified the issue on `Windows 10 x64 - 1.50.106`, I saw some discrepancies which are listed below:

1. When VPN is connected I have observed the below:
   - confirmed I saw the modal dialog about Windows' `Smart Multi-Homed Name Resolution` and our Brave-helper-service [ **_I saw this modal twice in clean profiles, this modal is not shown everytime when VPN is disconnected and connected in win 11, but in win 10 the modal is shown everytime when VPN is disconnected and connected_** ]
  - open Task Manager and ensured Brave VPN Helper service is running Details tab in task manager VPN is connected
  - ensured BraveberaVpnService is listed under the Service tab in task manager
  - ensured Brave VPN Helper is running under background apps when VPN is connected
1. When VPN is connected, the https://www.browserleaks.com/dns doesn't leak the user ISP DNS server as expected
3. **_When VPN is connected, open brave://settings/security, the `Use secure DNS` settings is NOT locked by BraveVPN_**

**Comment by GeetaSarvadnya** [comment]
Verified the issue on `Windows 11 x64 - 1.50.106`, I saw some discrepancies which are listed below:

1. When VPN is connected I have observed the below:
   - confirmed I saw the modal dialog about Windows' `Smart Multi-Homed Name Resolution` and our Brave-helper-service [ **_I saw this modal twice in clean profiles, this modal is not shown everytime when VPN is disconnected and connected in win 11, but in win 10 the modal is shown everytime when VPN is disconnected and connected_** ]
  - open Task Manager and ensured Brave VPN Helper service is running Details tab in task manager VPN is connected
  - ensured BraveberaVpnService is listed under the Service tab in task manager
  - ensured Brave VPN Helper is running under background apps when VPN is connected
1. When VPN is connected, the https://www.browserleaks.com/dns doesn't leak the user ISP DNS server as expected
3. **_When VPN is connected, open brave://settings/security, the `Use secure DNS` settings is NOT locked by BraveVPN_**
